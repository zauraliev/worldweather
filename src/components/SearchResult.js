import React, { Component } from "react";
import SearchResultRow from "./SearchResultRow";
import ArrowSVG from "./ArrowSVG";
class SearchResult extends Component {
  clear = e => {
    e.preventDefault();
    this.props.updateState({
      location: "",
      temperature: "",
      humidity: "",
      conditions: "",
      touched: {
        city: false,
        country: false
      },
      city: "",
      country: "",
      showResult: false,
      value: "",
      countryCode: ""
    });
  };
  render() {
    const showResult = this.props.showResult;
    const isLoading = this.props.isLoading;
    return (
      <div className="search-result ">
        <div className="row justify-content-center">
          <div className="col-4">
            <i
              className={"mt-5 " + (isLoading ? "fa fa-circle-o-notch fa-spin" : "")}
              style={{ fontSize: "60px" }}
            />
            <div className={"arrow " + (showResult ? "d-none" : "")}>
              <ArrowSVG />
              <p>Enter a location above to get started</p>
            </div>
            <div className={!showResult ? "d-none" : ""}>
              <div className="row-bordered-s">
                <SearchResultRow label="Location" value={this.props.location} />
              </div>
              <div className="row-bordered-o">
                <SearchResultRow label="Temperature" value={this.props.temperature} />
              </div>
              <div className="row-bordered-s">
                <SearchResultRow label="Humidity" value={this.props.humidity} />
              </div>
              <div className="row-bordered-o">
                <SearchResultRow label="Conditions" value={this.props.conditions} />
              </div>
              <button className="btn btn-clear btn-md btn-block mt-3" onClick={this.clear}>
                <span className="btn-text">Clear</span>
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export default SearchResult;
