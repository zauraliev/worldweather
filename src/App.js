import React, { Component } from "react";
import Header from "./components/Header";
import SearchBox from "./components/SearchBox";
import SearchResult from "./components/SearchResult";

class App extends Component {
  state = {
    location: "",
    temperature: "",
    humidity: "",
    conditions: "",
    touched: {
      city: false,
      country: false
    },
    city: "",
    country: "",
    showResult: false,
    isLoading: false,
    value: "",
    autocompleteData: []
  };
  updateState = state => {
    this.setState(state);
  };
  capitalize = ([first, ...rest]) => first.toUpperCase() + rest.join("").toLowerCase();

  calculateTemp = temp => {
    const tempSign = "Â°F";
    const dayTimeTemp = Math.round(1.8 * (temp - 273.15) + 32);

    return `${dayTimeTemp}${tempSign}`;
  };

  componentDidMount = async () => {
    // This part is for Autocomplete inputs for SearchBox
    // will continue work on it later
    const countriesData = require("./api/countries");
    const citiesData = require("./api/country_codes_cities");

    const countriesWithCodes = [];

    if (countriesData) {
      for (let i = 0; i < countriesData.length; i++) {
        let obj = {};
        obj.label = countriesData[i].name;
        obj.value = countriesData[i].alpha2Code;
        countriesWithCodes.push(obj);
      }

      var citiesClone = JSON.parse(JSON.stringify(citiesData));

      this.setState({
        countries: countriesWithCodes,
        autocompleteData: countriesWithCodes,
        cities: citiesClone
      });
    }
  };

  render() {
    return (
      <div className="wrapper">
        <div className="main">
          <div className="container-fluid">
            <div className="row justify-content-center align-items-center">
              <div className="col-4">
                <div className="form-group">
                  <Header />
                </div>
                <SearchBox
                  city={this.state.city}
                  country={this.state.country}
                  error={this.state.error}
                  touched={this.state.touched}
                  countries={this.state.countries}
                  updateState={this.updateState}
                  value={this.state.value}
                  autocompleteData={this.state.autocompleteData}
                  capitalize={this.capitalize}
                  calculateTemp={this.calculateTemp}
                />
              </div>
            </div>
          </div>
          <SearchResult
            location={this.state.location}
            temperature={this.state.temperature}
            humidity={this.state.humidity}
            conditions={this.state.conditions}
            error={this.state.error}
            showResult={this.state.showResult}
            isLoading={this.state.isLoading}
            updateState={this.updateState}
          />
        </div>
      </div>
    );
  }
}

export default App;
