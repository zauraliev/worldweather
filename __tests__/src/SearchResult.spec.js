import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import SearchResult from "../../src/components/SearchResult";
import App from "../../src/App";

describe("SearchResult", () => {
  it("clear has to clear form and state", () => {
    const propsMock = {
      location: "",
      temperature: "",
      humidity: "",
      conditions: "",
      error: undefined,
      showResult: true,
      isLoading: false,
      updateState: jest.fn()
    };

    const app = shallow(<App />).instance();
    const updateStateSpy = jest.fn(() => {
      app.setState(propsMock);
    });

    const wrapper = shallow(<SearchResult updateState={updateStateSpy} />).instance();

    const preventDefaultSpy = jest.fn();
    const resetFormSpy = jest.fn();

    wrapper.clear({
      preventDefault: preventDefaultSpy,
      resetForm: resetFormSpy,
      updateState: updateStateSpy
    });

    expect(preventDefaultSpy.mock.calls.length).toBe(1);
    expect(resetFormSpy.mock.calls.length).toBe(0);
    expect(updateStateSpy.mock.calls.length).toBe(1);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<SearchResult />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
