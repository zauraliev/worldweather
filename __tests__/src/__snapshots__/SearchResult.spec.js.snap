// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult renders correctly 1`] = `
<div
  className="search-result "
>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-4"
    >
      <i
        className="mt-5 "
        style={
          Object {
            "fontSize": "60px",
          }
        }
      />
      <div
        className="arrow "
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
        >
          <svg
            height="100px"
            viewBox="0 0 150 300"
            y="20%"
          >
            <polygon
              className="polygon"
              points="70 25,5 80,10 85,70 30,135 85,140 80"
            />
            <polygon
              className="polygon"
              points="70 55,10 107.5,12.5 110,70 60,132.5 110,135.5 107.5"
            />
          </svg>
        </svg>
        <p>
          Enter a location above to get started
        </p>
      </div>
      <div
        className="d-none"
      >
        <div
          className="row-bordered-s"
        >
          <div
            className="row"
          >
            <div
              className="col-md-3 text-lg-left"
            >
              <label
                className="font-weight-bold list-header mr-1"
              >
                Location
              </label>
            </div>
            <div
              className="col-md-6 text-lg-left"
            >
              <span />
            </div>
          </div>
        </div>
        <div
          className="row-bordered-o"
        >
          <div
            className="row"
          >
            <div
              className="col-md-3 text-lg-left"
            >
              <label
                className="font-weight-bold list-header mr-1"
              >
                Temperature
              </label>
            </div>
            <div
              className="col-md-6 text-lg-left"
            >
              <span />
            </div>
          </div>
        </div>
        <div
          className="row-bordered-s"
        >
          <div
            className="row"
          >
            <div
              className="col-md-3 text-lg-left"
            >
              <label
                className="font-weight-bold list-header mr-1"
              >
                Humidity
              </label>
            </div>
            <div
              className="col-md-6 text-lg-left"
            >
              <span />
            </div>
          </div>
        </div>
        <div
          className="row-bordered-o"
        >
          <div
            className="row"
          >
            <div
              className="col-md-3 text-lg-left"
            >
              <label
                className="font-weight-bold list-header mr-1"
              >
                Conditions
              </label>
            </div>
            <div
              className="col-md-6 text-lg-left"
            >
              <span />
            </div>
          </div>
        </div>
        <button
          className="btn btn-clear btn-md btn-block mt-3"
          onClick={[Function]}
        >
          <span
            className="btn-text"
          >
            Clear
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
